modules/fedimint-ln/src/db.rs

```diff
@@ -118,3 +118,143 @@ impl_db_lookup!(
+                                    .await;
+                                let num_contracts = contracts.len();
+                                for contract in contracts {
+                                    contract.expect("Error deserializing contract");
+                                }
+                                migrated_pairs.insert(DbKeyPrefix::Contract as u8, num_contracts);
+                            }
+                            DbKeyPrefix::AgreedDecryptionShare => {
+                                let agreed_decryption_shares = dbtx
+                                    .find_by_prefix(&AgreedDecryptionShareKeyPrefix)
+                                    .await
+                                    .collect::<Vec<_>>()
+                                    .await;
+                                let num_shares = agreed_decryption_shares.len();
+                                for share in agreed_decryption_shares {
+                                    share.expect("Error deserializing AgreedDecryptionShare");
+                                }
+                                migrated_pairs
+                                    .insert(DbKeyPrefix::AgreedDecryptionShare as u8, num_shares);
+                            }
+                            DbKeyPrefix::ContractUpdate => {
+                                let contract_updates = dbtx
+                                    .find_by_prefix(&ContractUpdateKeyPrefix)
+                                    .await
+                                    .collect::<Vec<_>>()
+                                    .await;
+                                let num_updates = contract_updates.len();
+                                for update in contract_updates {
+                                    update.expect("Error deserializing ContractUpdate");
+                                }
+                                migrated_pairs
+                                    .insert(DbKeyPrefix::ContractUpdate as u8, num_updates);
+                            }
+                            DbKeyPrefix::LightningGateway => {
+                                let gateways = dbtx
+                                    .find_by_prefix(&LightningGatewayKeyPrefix)
+                                    .await
+                                    .collect::<Vec<_>>()
+                                    .await;
+                                let num_gateways = gateways.len();
+                                for gateway in gateways {
+                                    gateway.expect("Error deserializing LightningGateway");
+                                }
+                                migrated_pairs
+                                    .insert(DbKeyPrefix::LightningGateway as u8, num_gateways);
+                            }
+                            DbKeyPrefix::Offer => {
+                                // TODO: Offers are temporary database entries
```

Comment:
That's why I think module developer should write manually their own logic to populate the database.

Migrations must work in any state, even if federation got shut down/crashed in them middle of something weird.

Reply:
Outdated, backups are no longer made from running the integration tests